federation-js/src/composition/validate/postComposition/__tests__/keyFieldsMissingOnBase.test.ts

```diff
@@ -46,6 +46,46 @@ describe('keyFieldsMissingOnBase', () => {
     expect(warnings).toHaveLength(0);
   });
 
+  it('returns no warnings with proper @key usage for provided field', () => {
+    const serviceA = {
+      // FIXME: add second key "upc" when duplicate directives are supported
+      // i.e. @key(fields: "sku") @key(fields: "upc")
+      typeDefs: gql`
+        type Product @key(fields: "sku") {
+          sku: String!
+          upc: String!
+          color: Color!
+        }
+
+        type Color {
+          id: ID!
+          value: String!
+        }
+      `,
+      name: 'serviceA',
+    };
+
+    const serviceB = {
+      typeDefs: gql`
+        type ProductCatalog @key(fields: "products { sku }"){
+          products: [Product!]! @provides(fields: "price")
+        }
+        extend type Product {
```

Comment:
Kind of a nit: I know this won't cause any validation errors as-is since we're isolating the validations in these test cases. But probably best if our test cases are mostly valid. I know it's like this above as well.
```suggestion
        extend type Product @key(fields: "sku") {
```